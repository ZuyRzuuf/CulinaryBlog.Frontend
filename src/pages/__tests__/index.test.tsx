import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import IndexPage from "../index";

describe('IndexPage', () => {
    it('renders the hero section with title and subtitle', () => {
        render(<IndexPage />);
        const titleElement = screen.getByText(/Pan Przy Garach/i);
        const subtitleElement = screen.getByText(/Discover delicious recipes and culinary inspiration/i);

        expect(titleElement).toBeInTheDocument();
        expect(subtitleElement).toBeInTheDocument();
    });

    it('renders the Featured Recipes section', () => {
        render(<IndexPage />);
        const featuredRecipesHeading = screen.getByText(/Featured Recipes/i);

        expect(featuredRecipesHeading).toBeInTheDocument();
    });

    it('renders the Recent Articles section', () => {
        render(<IndexPage />);
        const recentArticlesHeading = screen.getByText(/Recent Articles/i);

        expect(recentArticlesHeading).toBeInTheDocument();
    });

    it('renders the Subscribe to Our Newsletter section', () => {
        render(<IndexPage />);
        const subscribeHeading = screen.getByText(/Subscribe to Our Newsletter/i);
        const subscribeDescription = screen.getByText(/Get the latest recipes, articles, and updates delivered to your inbox./i);

        expect(subscribeHeading).toBeInTheDocument();
        expect(subscribeDescription).toBeInTheDocument();
    });
})
