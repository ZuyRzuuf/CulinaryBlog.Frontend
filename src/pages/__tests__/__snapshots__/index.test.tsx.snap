// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="bg-gray-800 text-white p-4"
  >
    <nav>
      <a
        href="/"
      >
        <span
          class="mx-2 cursor-pointer"
        >
          Home
        </span>
      </a>
      <a
        href="/recipes"
      >
        <span
          class="mx-2 cursor-pointer"
        >
          Recipes
        </span>
      </a>
      <a
        href="/articles"
      >
        <span
          class="mx-2 cursor-pointer"
        >
          Articles
        </span>
      </a>
      <a
        href="/about"
      >
        <span
          class="mx-2 cursor-pointer"
        >
          About
        </span>
      </a>
      <a
        href="/contact"
      >
        <span
          class="mx-2 cursor-pointer"
        >
          Contact
        </span>
      </a>
    </nav>
  </header>
  <main
    class="p-4"
  >
    <section
      class="bg-cover bg-center h-96 relative text-white"
      style="background-image: url(path/to/hero-image.jpg);"
    >
      <div
        class="absolute inset-0 bg-black opacity-50"
      />
      <div
        class="container mx-auto px-4 h-full flex items-center justify-center"
      >
        <div>
          <h1
            class="text-4xl font-bold"
          >
            Pan Przy Garach
          </h1>
          <p
            class="text-xl mt-2"
          >
            Discover delicious recipes and culinary inspiration
          </p>
        </div>
      </div>
    </section>
    <section
      class="my-16"
    >
      <h2
        class="text-2xl font-bold mb-8"
      >
        Featured Recipes
      </h2>
    </section>
    <section
      class="my-16"
    >
      <h2
        class="text-2xl font-bold mb-8"
      >
        Recent Articles
      </h2>
    </section>
    <section
      class="my-16 bg-gray-100 p-8"
    >
      <h2
        class="text-2xl font-bold mb-4"
      >
        Subscribe to Our Newsletter
      </h2>
      <p
        class="mb-4"
      >
        Get the latest recipes, articles, and updates delivered to your inbox.
      </p>
    </section>
  </main>
  <footer
    class="bg-gray-800 text-white p-4"
  >
    <p>
      Â© 2023 PanPrzyGarach
    </p>
  </footer>
</DocumentFragment>
`;
